outputs.tf

```diff
@@ -1,25 +1,29 @@
 output "id" {
-  value       = join("", azurerm_network_security_group.nsg.*.id)
+  value       = join("", azurerm_network_security_group.nsg[*].id)
```

Comment:
remove join for ouputs that are only used for single resource creation. Use [0] instead of [*]. Pls do for all outputs.